source/backend/database/models/completedRecipesModel.js

```diff
@@ -13,4 +13,31 @@ async function getCompletedRecipes(userId){
   return completedRecipes;
 }
 
-module.exports = { getCompletedRecipes };
\ No newline at end of file
+/**
+ *  Adds a new recipe to the user's completed list by recipeID
+ * @param recipeId
+ * @returns {Promise<awaited Knex.QueryBuilder<TRecord, TResult>>}
+ */
+ async function addNewRecipe(recipeId, userId){
+  
+  await db.transaction(async (transaction) => {
+    try {
+      await db('completedRecipes')
+            .transacting(transaction)
+            .where({userId, recipeId})
+            .insert();
+      await transaction.commit();
+    } catch (err) {
+      console.log(err);
+      await transaction.rollback();
+    }
+  });
+  
+  const updatedRecipeList = await db('completedRecipes')
+      .where('updatedRecipeList.recipeId', recipeId)
+      .insert({ recipeId })
+      .select('recipes.*');
+    return updatedRecipeList;
+}

```

Comment:
I think this might be all you need for this function, although I'm not too sure about the `.where('updatedRecipeList.recipeId'...`
Can we just do:
```
const updatedRecipeList = await db('completedRecipes')
             .insert({userId, recipeId});
return updatedRecipeList
```
Would that work? 

I think there is something wrong with your code because you are selecting all recipes from recipes table where the updatedRecipeList's recipe Id and recipeId match completedRecipes, then insert the recipeId? I'm not sure if I'm reading this correctly, but I think there might be some errors. 

Have you been able to test it?

Reply:
Yes, I didn't need to select tall the recipes and I don't think I need the .where() either. Thanks for pointing both of these out. I have yet to test the implementation.